Packages/Timeline/Sources/Timeline/TimelineViewModel.swift

```diff
@@ -100,6 +111,44 @@ class TimelineViewModel: ObservableObject, StatusesFetcher {
       print("timeline parse error: \(error)")
     }
   }
+  
+  func digestTimeline() async -> [Status] {
+    // fetch all the statuses until specified time (last 24 hours) and then run the algorithm
+    let hoursAgo = 24
+
+    // Get as many posts as possible (max seems to be 400 though https://github.com/mastodon/mastodon/issues/2301)
+    let earlyDate = Calendar.current.date(
+      byAdding: .hour,
+      value: -hoursAgo,
+      to: Date()
+    )!
+    let timestamp = ISO8601DateFormatter().string(from: earlyDate)
+    let allStatuses = await fetchNewPages(minId: timestamp, maxPages: 400)
```

Comment:
So the API accept a minId that is in a form of a timestamp and not an id of a Status? 
Also 400 might be bit much considering that page contains 40 status? With 10 you would already get 400 status of data. Not sure there.

Reply:
I just put "a huge number" there. But the Mastodon is limited to max 400 toots for the home timeline API call.